libevmasm/RuleList.h

```diff
@@ -89,6 +89,16 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleList(
 			u256 mask = (u256(1) << testBit) - 1;
 			return u256(boost::multiprecision::bit_test(B.d(), testBit) ? B.d() | ~mask : B.d() & mask);
 		}, false},
+		{{Instruction::SHL, {A, B}}, [=]{
```

Comment:
@chriseth do we want to introduce `evmVersion` in the rule list or is it safe to have this merged (after some cleanup/squashing)?

Reply:
I think it should be safe. Let's add a comment.

Reply:
Are you sure about the order of the operands?

Reply:
Please add a test.

Reply:
(is that possible?)

Reply:
This is blocked by the same problem as in #3797: need to build a constantinople supporting version of `eth` and run endtoend tests with that target.